Lib/test/test_remote_pdb.py

```diff
@@ -78,6 +79,697 @@ def get_output(self) -> List[dict]:
         return results
 
 
+class MockDebuggerSocket:
+    """Mock file-like simulating a connection to a _RemotePdb instance"""
+
+    def __init__(self, incoming):
+        self.incoming = iter(incoming)
+        self.outgoing = []
+        self.buffered = bytearray()
+
+    def write(self, data: bytes) -> None:
+        """Simulate write to socket."""
+        self.buffered += data
+
+    def flush(self) -> None:
+        """Ensure each line is valid JSON."""
+        lines = self.buffered.splitlines(keepends=True)
+        self.buffered.clear()
+        for line in lines:
+            assert line.endswith(b"\n")
+            self.outgoing.append(json.loads(line))
+
+    def readline(self) -> bytes:
+        """Read a line from the prepared input queue."""
+        # Anything written must be flushed before trying to read,
+        # since the read will be dependent upon the last write.
+        assert not self.buffered
+        try:
+            item = next(self.incoming)
+            if not isinstance(item, bytes):
+                item = json.dumps(item).encode()
+            return item + b"\n"
+        except StopIteration:
+            return b""
+
+    def close(self) -> None:
+        """No-op close implementation."""
+        pass
+
+
+class PdbClientTestCase(unittest.TestCase):
+    """Tests for the _PdbClient class."""
+
+    def do_test(
+        self,
+        *,
+        incoming,
+        simulate_write_failure=False,
```

Comment:
I feel like having two arguments for rarely used failures is a bit luxury. Maybe `simulate_failure=None or "write" or "flush"`?